### PR TITLE
Leaf 4

### PR DESCRIPTION
**Pull request by 0xTim**
Prepping for a Leaf 4 release

### CODE REVIEW

File: 4.0/docs/leaf/custom-tags.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+To demonstrate this, let's take a look at creating a custom tag `#now` that prints the current timestamp. The tag will also support a single, optional parameter for specifying the date format.
+
+## Tag Renderer
+
+First create a class called `NowTag` and conform it to `LeafTag`.
+
+```swift
+final class NowTag: LeafTag {
+    init() { }
```
Comment (siemensikkema) [inline 4.0/docs/leaf/custom-tags.md:13]: Shouldn't we encourage structs?

```suggestion
struct NowTag: LeafTag {
```

File: 4.0/docs/leaf/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+app.views.use(.leaf)
+```
+
+This tells Vapor to use the `LeafRenderer` when you call `req.view` in your code.
+
+## Folder Structure
+
+Once you have configured Leaf, you will need to ensure you have a `Views` folder to store your `.leaf` files in. By default, Leaf expects the views folder to be a `./Resources/Views` relative to your project's root.
+
+You will also likely want to enable Vapor's [`FileMiddleware`](https://api.vapor.codes/vapor/latest/Vapor/Classes/FileMiddleware.html) to serve files from your `/Public` folder.
```
Comment (siemensikkema) [inline 4.0/docs/leaf/getting-started.md:48]:
```suggestion
You will also likely want to enable Vapor's [`FileMiddleware`](https://api.vapor.codes/vapor/latest/Vapor/Classes/FileMiddleware.html) to serve files from your `/Public` folder if you plan on serving Javascript and CSS files for instance.
```

File: 4.0/docs/leaf/overview.md [hunk 3]
```diff
@@ -0,0 +1,258 @@
+# Leaf Overview
+
+Leaf is a powerful templating language with Swift-inspired syntax. You can use it to generate dynamic HTML pages for a front-end website or generate rich emails to send from an API.
```
Comment (siemensikkema) [inline 4.0/docs/leaf/overview.md:3]: ok to repeat the same sentence from getting-started?

File: 4.0/docs/leaf/overview.md [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+```
+Planets:
+- Venus
+- Earth
+- Mars
+```
+
+### Extending templates
+
+Leaf’s `#extend` tag allows you to copy the contents of one template into another. When use this, you should always omit the template file's .leaf extension.
```
Comment (siemensikkema) [inline 4.0/docs/leaf/overview.md:152]:
```suggestion
Leaf’s `#extend` tag allows you to copy the contents of one template into another. When using this, you should always omit the template file's .leaf extension.
```

File: 4.0/docs/leaf/overview.md [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+
+Extending is useful for copying in a standard piece of content, for example a page footer, advert code or table that's shared across multiple pages:
+
+```leaf
+#extend("footer")
+```
+
+This tag is also useful for building one template on top of another. For example, you might have a master.leaf file that includes all the code required to lay out your website – HTML structure, CSS and JavaScript – with some gaps in place that represent where page content varies.
```
Comment (grantjbutler) [inline 4.0/docs/leaf/overview.md:160]: Maybe switch to more inclusive language for the naming of the main layout file?

```suggestion
This tag is also useful for building one template on top of another. For example, you might have a layout.leaf file that includes all the code required to lay out your website – HTML structure, CSS and JavaScript – with some gaps in place that represent where page content varies.
```

File: 4.0/docs/leaf/getting-started.md (hunk 2 above)
Comment (grundoon) [reply 4.0/docs/leaf/getting-started.md:48]: Feels a bit awkward to me, perhaps something along these lines:

If you plan on including static content such as Javascript or CSS files within your rendered views, you will also likely want to enable Vapor's [`FileMiddleware`](https://api.vapor.codes/vapor/latest/Vapor/Classes/FileMiddleware.html) to serve these from your `/Public` folder.

File: 4.0/docs/leaf/custom-tags.md (hunk 1 above)
Comment (0xTim) [reply 4.0/docs/leaf/custom-tags.md:13]: Yes good point

File: 4.0/docs/leaf/overview.md (hunk 5 above)
Comment (0xTim) [reply 4.0/docs/leaf/overview.md:160]: Yes I agree, good spot!

File: 4.0/docs/leaf/overview.md (hunk 3 above)
Comment (0xTim) [reply 4.0/docs/leaf/overview.md:3]: Yeah I think so. It's a copy from Leaf 3

File: 4.0/docs/leaf/overview.md [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+#if(contains(planets, "Earth")):
+    Earth is here!
+#elseif
+    Earth is not in this array.
+#endif
+```
+
+#### `#date`
+
+The `#date` tag formats dates into a readable string.
```
Comment (Joannis) [inline 4.0/docs/leaf/overview.md:244]: What's the default date format?

File: 4.0/docs/leaf/overview.md [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+
+The `#uppercased` tag uppercases all letters in a string.
+
+```leaf
+#uppercased(name)
+```
+
+#### `#capitalized`
+
+The `#capitalized` tag uppercases the first letter of any string.
```
Comment (Joannis) [inline 4.0/docs/leaf/overview.md:224]: Does it lowercase the rest, or keep it the same? Does it capitalise each word, or each string as a whole?

Comment (0xTim) [reply 4.0/docs/leaf/overview.md:224]: Good shout

File: 4.0/docs/leaf/overview.md (hunk 6 above)
Comment (0xTim) [reply 4.0/docs/leaf/overview.md:244]: Good point

**Comment by 0xTim** [comment]
@Joannis changes made

**Review by Joannis — APPROVED** [review]

**Pull request merged**